DOC-1770: TinyMCE 6.1.2 release notes (#2486)
tinymce/tinymce-docs: modules/ROOT/pages/6.1.2-release-notes.adoc

* xref:bug-fixes[Bug fixes]
* xref:known-issues[Known issues]

[[bug-fixes]]
== Bug fixes

=== Invalid special elements were not cleaned up correctly during sanitization

In {productname} 5.x, loading data that contained invalid special elements resulted in the invalid elements and their contents being removed before the data was placed in a {productname} document.

In {productname} 6, a change to the DomParser introduced a regression: invalid elements were still filtered out, but their contents were placed in the {productname} document as plain text.

The {productname} 6.1.2 update reverts this regression. When input data is being parsed by {productname} 6.1.2, invalid elements *and* their contents are both removed, as happened previously and is expected.

=== An exception was thrown when deleting all content if the start or end of the document had a `contenteditable="false"`

If the first or last element in a {productname} document had a `contenteditable="false"` property and the entire {productname} document was selected (eg, *Edit > Select All* was chosen), pressing the Delete or Backspace key resulted in a console error: `Uncaught Error: Node cannot be null or undefined`.

> With this update, the exception is no longer thrown in this circumstance. In {productname} 6.1.2, selecting all of a {productname} document that includes a `contenteditable="false"` property at the beginning or end of the document and deleting the selected contents produces no console error.

Reviewer: Just wondering if saying 'element' rather than 'property' make a bit more sense for the sentence. No strong opinion though.
```suggestion
With this update, the exception is no longer thrown in this circumstance. In {productname} 6.1.2, selecting all of a {productname} document that includes a `contenteditable="false"` element at the beginning or end of the document and deleting the selected contents produces no console error.
```

[[toolbar-button-highlights]]
=== When a sidebar was opened using the `sidebar_show` option, its associated toolbar button was not highlighted

If a {productname} instance is set to initialize with both a sidebar and an associated toolbar button, the toolbar button should be highlighted.

If, for example, a {productname} instance 

> 1. presents a toolbar which includes a Show comments button — `toolbar: "showcomments",` and

Reviewer 2: ```suggestion
1. presents a toolbar which includes a Show comments button — `toolbar: "showcomments"` and
```
Don't think that the comma should be here

> 2. has Comments set to appear at initialization — `sidebar_show: 'showcomments',` 

Reviewer 2: ```suggestion
2. has Comments set to appear at initialization — `sidebar_show: "showcomments"` 
```

the _Show Comments_ button should be highlighted on {productname} initialization.

> However, in {productname} 6.1, if a sidebar and its associated toolbar button were both set to appear on initialization, the associated toolbar button did not hightlight.

Reviewer 2: ```suggestion
However, in {productname} 6.1, if a sidebar and its associated toolbar button were both set to appear on initialization, the associated toolbar button was not highlighted.
```

= TinyMCE 6.1.2
:navtitle: TinyMCE 6.1.2
:description: Release notes for TinyMCE 6.1.2
:keywords: releasenotes, new, changes, bugfixes

include::partial$misc/admon-releasenotes-for-stable.adoc[]

[[overview]]
== Overview

{productname} 6.1.2 was released for {enterpriseversion} and {cloudname} on Wednesday, August 3^rd^, 2022. It includes {productname} 6.1.2. These release notes provide an overview of the changes for {productname} 6.1.2:

* xref:bug-fixes[Bug fixes]
* xref:known-issues[Known issues]

[[bug-fixes]]
== Bug fixes

=== Invalid special elements were not cleaned up correctly during sanitization

> In {productname} 5.x, loading data that contained invalid elements resulted in the invalid elements and their contents being removed before the data was placed in a {productname} document.

Reviewer 3: ```suggestion
In {productname} 5.x, loading data that contained invalid special elements resulted in the invalid elements and their contents being removed before the data was placed in a {productname} document.
```
This only happens for special elements, not all invalid elements. For all others they are just unwrapped instead.

In {productname} 6, a change to the DomParser introduced a regression: invalid elements were still filtered out, but their contents were placed in the {productname} document as plain text.

The {productname} 6.1.2 update reverts this regression. When input data is being parsed by {productname} 6.1.2, invalid elements *and* their contents are both removed, as happened previously and is expected.

=== An exception was thrown when deleting all content if the start or end of the document had a `contenteditable="false"`

If the first or last element in a {productname} document had a `contenteditable="false"` property and the entire {productname} document was selected (eg, *Edit > Select All* was chosen), pressing the Delete or Backspace key resulted in a console error: `Uncaught Error: Node cannot be null or undefined`.

With this update, the exception is no longer thrown in this circumstance. In {productname} 6.1.2, selecting all of a {productname} document that includes a `contenteditable="false"` property at the beginning or end of the document and deleting the selected contents produces no console error.

[[show-comments-toolbar-button-highlights]]

> === When a sidebar was opened using the `sidebar_show` option, its associated toolbar button was not highlighted

Reviewer 3: This heading starts out about being about all sidebars, but then gets very comments specific below. While comments was a case, I'm not sure if it should be the only focus below?

Author: Good point. I’ve pushed a re-write of this, and associated edits to link text.

However, in {productname} 6.1, if the Comments sidebar is set to show at initialization — `sidebar_show: 'showcomments',` — the Show comments button did not highlight.

With the {productname} 6.1.2 update, this visual misbehavior has been corrected. When the Comments sidebar is set to show on initialization, the Show comments button, if present, is now highlighted.

NOTE: This was a known issue in the {productname} 6.1 release.

=== When converting a URL to a link, the `autolink` plugin did not fire an `ExecCommand` event

The `autolink` plugin converts typed or pasted strings that match its criteria for being URLs into clickable blocks. It acts when a word-boundary character (such as a space or a closing bracket) is typed.

In {productname} 6, however, the plugin did not fire an `ExecCommand` event.

Consequently no record of the link creation was available.

This update corrects this. In {productname} 6.1.2, the `autolink` plugin fires an `ExecCommand` event, as expected.

=== Worked around a Firefox bug that results in cookies not being available inside the editor content

> As part of the changes made for version 6, {productname} switched from the now-partly deprecated `iframe.contentDocument.write()` browser API to the newer, `srcDoc` attribute.

Reviewer 3: ```suggestion
As part of the changes made for version 6, {productname} switched from the now-partly deprecated `iframe.contentDocument.write()` browser API to the newer, `srcdoc` attribute.
```
This is the actual attribute name 🤷

In doing so, however, {productname} became subject to a bug in Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1741489[Firefox bug 1741489: cookies from parent window aren’t sent in iframe via srcDoc]. This Firefox bug causes cookies to not be available to the {productname} editor.

{productname} 6.1.2 includes a workaround: if Firefox is the host browser, iframe content is loaded using the older `iframe.contentDocument.write()` API. This ensures set cookies are available to {productname}.

> NOTE: This is a Firefox-specific workaround. When other browsers are the host, {productname} continues to use the, newer, `srcDoc` attribute.

Reviewer 3: ```suggestion
NOTE: This is a Firefox-specific workaround. When other browsers are the host, {productname} continues to use the, newer, `srcdoc` attribute.
```

[[pre-content-now-correctly-merges]]

> === Pasting `<pre>` content into a `<pre>` block that had inline styles or was `noneditable` now correctly merges with the surrounding content.

Reviewer 3: ```suggestion
=== Pasting `<pre>` content into a `<pre>` block that had inline styles or was `noneditable` now correctly merges with the surrounding content
```
The headings above don't have fullstops 🤷

Previously, when a `<pre>` block was pasted into a `noneditable` block, or was pasted into a `<pre>` block that also contained content with inline styles applied (such as `<strong>`), the pasted-in block did not merge correctly.

When pasted into `<strong>` content, the pasted-in `<pre>` block split the text at the paste point.

When pasted into a `noneditable` block it replaced the block with standard text.

In {productname} 6.1.2, pasting `<pre>` blocks merges with surrounding content as expected: it takes on the inline style when pasted into content with inline styles applied; and it retains its `<pre>` tags when pasted on to a `noneditable` block.

NOTE: This was a known issue in the {productname} 6.1 release.

> === After a `codesample` was pasted, the insertion point was placed incorrectly. #TINY-8861

Reviewer 3: ```suggestion
=== After a `codesample` was pasted, the insertion point was placed incorrectly
```

> Previously, if a `codesample` block (or other block with a `contenteditable="false"` attribute applied) was copied and then pasted, the insertion point was, incorrectly, placed inside the `contenteditable="false"` block.

Reviewer 3: ```suggestion
Previously, if a `codesample` block (or other pre block with a `contenteditable="false"` attribute applied) was copied and then pasted, the insertion point was, incorrectly, placed inside the `contenteditable="false"` block.
```
This only applied to `<pre>` blocks, not all blocks.
